### PR TITLE
TST: Use `sort` fixture in more places

### LINKED ISSUE #32183: Use `sort` fixture in more places
This is a follow-up on [this review comment by jbrockmendel](https://github.com/pandas-dev/pandas/pull/32046#discussion_r382697823).

There's a `sort` fixture in `tests.indexes.interval.test_setops`, which could be used in other places as well. For instance in `test_difference_base` in `pandas/tests/indexes/common.py`.

There are probably many more places where it could be used.

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
- [x] closes #32183
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
Lgtm @simonjayhawkins

**Comment by simonjayhawkins** [comment]
Thanks @SaturnFromTitan 

In order to close the issue can you do test_union_noncomparable in pandas\tests\indexes\ranges\test_setops.py as well.

There are several cases of @pytest.mark.parametrize("sort", [True, False]) as well. so having this fixture in top-level conftest maybe confusing.

for instance groupby where the default is True, unlike intersection where the default is False.

so is the None value in the fixture effectively True.

would tests break if the fixture was params=[None, True, False]?

**Comment by SaturnFromTitan** [comment]
Yes the tests break when `True` is added to the fixture params:
```
ValueError: The 'sort' keyword only takes the values of None or False; True was passed.
```

I'll move the fixture to pandas/tests/indexes/conftest.py instead and adjust the docstring

File: pandas/tests/indexes/conftest.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import pytest
+
+
+@pytest.fixture(params=[None, False])
+def sort(request):
+    """
+    Valid values for 'sort' parameter used in a variety Index methods.
```
Comment (simonjayhawkins) [inline pandas/tests/indexes/conftest.py]: It's the setops? intersection, union, difference etc?

File: pandas/tests/indexes/conftest.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    Caution:
+        There are also "sort" fixtures with params [True, False] or [None, True, False].
+        For instance these might be used for DataFrame.groupby.
```
Comment (simonjayhawkins) [inline pandas/tests/indexes/conftest.py]: we have one for concat and append, but AFAICT not groupby.

**Comment by SaturnFromTitan** [comment]
@simonjayhawkins I updated the docstring again, please have another look

**Review by simonjayhawkins — APPROVED** [review]
Thanks @SaturnFromTitan changes lgtm.

**Pull request merged**